lib/evaluation/evaluationClientWrapper.ts

```diff
@@ -474,16 +481,53 @@ export class EvaluationClientWrapper {
       };
     }
 
-    // Find the wrapped VC based on the original VC
-    const originalVc = vcResult.value;
-    const wvc = wvp.vcs.find((wvc) => CredentialMapper.areOriginalVerifiableCredentialsEqual(wvc.original, originalVc));
+    // FIXME figure out possible types, can't see that in debug mode...
+    const isCredential = CredentialMapper.isCredential(vcResult.value as OriginalVerifiableCredential);
+    if (
+      !vcResult.value ||
+      (typeof vcResult.value === 'string' && !isCredential) ||
+      (typeof vcResult.value !== 'string' && !isCredential && !('verifiableCredential' in vcResult.value || 'vp' in vcResult.value))
+    ) {
+      return {
+        error: {
+          status: Status.ERROR,
+          tag: 'NoVerifiableCredentials',
+          message: `No verifiable credentials found at path "${descriptor.path}" for submission.descriptor_path[${descriptorIndex}]`,
+        },
+        wvc: undefined,
+      };
+    }
+
+    // When result is an array, extract the first Verifiable Credential from the array FIXME figure out proper types, can't see that in debug mode...
+    let originalVc;
+    if (isCredential) {
+      originalVc = vcResult.value;
+    } else if (typeof vcResult.value !== 'string') {
+      if ('verifiableCredential' in vcResult.value) {
+        originalVc = Array.isArray(vcResult.value.verifiableCredential)
+          ? vcResult.value.verifiableCredential[0]
+          : vcResult.value.verifiableCredential;
+        // FIXME this may be too lenient
+      } else if ('vp' in vcResult.value) {
```

Comment:
How would a vp end up in a VC? And more importantly why would we only take the first credentials from the jwt VP then? It can have multiple credentials

Reply:
I encountered that while debugging. But later I saw that OID4VC had a specific test for that, that was supposed to fail.